frame/vault/src/lib.rs

```diff
@@ -849,6 +849,25 @@ pub mod pallet {
 			Ok(lp_shares_value_amount)
 		}
 
+		fn do_calculate_lp_tokens_from_asset_amount(
+			vault_id: &T::VaultId,
+			vault: &VaultInfo<T>,
+			asset_amount: T::Balance,
+		) -> Result<BalanceOf<T>, DispatchError> {
+			let total_lp_issuance = T::Currency::total_issuance(vault.lp_token_id);
+			let aum = Self::assets_under_management(vault_id)?;
+
+			let a = <T::Convert as Convert<T::Balance, u128>>::convert(asset_amount);
```

Comment:
so you have functions which receives 3 numbers, converts and converts back?

Reply:
which is same as on lines 81x

Reply:
can we manage some code reuse?

Reply:
in this pr

Reply:
You're right, I have refactored the code with your suggestions, let me know if it's better now or if I should change something else too